langlib/lang.decimal/src/main/java/org/ballerinalang/langlib/decimal/Quantize.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.langlib.decimal;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+/**
+ * Native implementation of lang.decimal:quantize(decimal, decimal).
+ *
+ * @since 2201.1.0
+ */
+public class Quantize {
+    public static BDecimal quantize(BDecimal x, BDecimal y) {
+        BigDecimal quantizeValue = x.value().setScale(y.value().scale(), RoundingMode.HALF_EVEN);
+        //  The maximum number of digits in the significand(precision) of 128-bit decimal (radix 10)
+        //  floating-point number equals 34
+        if (quantizeValue.precision() > 34) {
```

Comment:
Even if this is correct, it is not a good practice to rely exactly on 34. We may have to find some other way. Let me also think about that. 

BigDecimal does not care about this right?

Reply:
Please check whether BigDecimal set a flag for these cases

Reply:
Didn't find such kind of a solution. 
Yeah, the main reason is that `BigDecimal` does not care about the number of digits in the significand(precision) value.
```ballerina
import ballerina/io;
public function main() {
    decimal|error k = decimal:fromString("234.65678877997979799979797997979797997979");
    io:println(k);
}
```
In ballerina, the above program result is `234.6567887799797979997979799797980`.  Here the number of digits in precision is equal to 34.
```java
BigDecimal a = new BigDecimal("234.65678877997979799979797997979797997979");
```
but in java, the result is `234.65678877997979799979797997979797997979`. Here number of digits in precision value not considered.
According to IEEE format, there can be a maximum of 34 digits in precision.

Reply:
Yes, I also checked, was unable to find a better solution. Define 34 as a constant and move the comment when defining it.